anaconda/pyanaconda/bootloader.py

```diff
@@ -1908,16 +1908,122 @@ def is_valid_stage2_device(self, device, linux=True, non_linux=False):
 
     write_config = BootLoader.write_config
 
-class MacEFIGRUB(EFIGRUB):
+class MacEFIGRUB(XenEFI):
+    """Special EFI handling for macOS HFS+ ESP partition.
+
+    Typical GRUB2 installations would execute the script
+    located at /usr/libexec/mactel-boot-setup which would
+    modify the HFS+ ESP files and bless the specified efi.
+
+    However, we are not using GRUB at this time which would
+    cause that script to exit earlier.  
+
+    In this class, we will execute the relevant commands
+    to symlink the efi file in the /System directory as well
+    the cfg file.  Lastly, macOS requires the bootable efi
+    file to be blessed.
+    """
+
     def mactel_config(self):
-        if os.path.exists(iutil.getSysroot() + "/usr/libexec/mactel-boot-setup"):
-            rc = iutil.execInSysroot("/usr/libexec/mactel-boot-setup", [])
-            if rc:
-                log.error("failed to configure Mac boot loader")
+        """Modifies the HFS+ ESP partition to be bootable.
+
+        Based on the /usr/libexec/mactel-boot-setup script, 
+        we will create two symlinks pointing to the Qubes
+        Xen version that is installed and configured for
+        the system.  We also need to run hfs-bless on
+        the specific EFI file we allow to be bootable.
+        """
+        efi_fil = "{}/{}".format(iutil.getSysroot() + self.config_dir, "xen.efi")
+        efi_cfg = "{}/{}".format(iutil.getSysroot() + self.config_dir, "xen.cfg")
+        sys_dir = iutil.getSysroot() + "/boot/efi/System/Library/Coreservices/"
+        bless_file = "{}/{}".format(self.config_dir, "xen.efi")
+
+        not_ready = False
+        if not os.path.exists(efi_fil):
+            log.error("mactel efi_fil not found: %s", efi_fil)
+            not_ready = True
+
+        if not os.path.exists(efi_cfg):
+            log.error("mactel efi_cfg not found: %s", efi_cfg)
+            not_ready = True
+
+        if not os.path.exists(sys_dir):
+            log.error("mactel sys_dir not found: %s", sys_dir)
+            not_ready = True
+            
+        if not_ready:
+            log.error("mactel cannot continue with mactel_config. see log for details.")
+            return
+
+        sys_dir_boot_fil = os.path.normpath(sys_dir + "boot.efi")
+        try:
+            os.symlink(efi_fil, sys_dir_boot_fil)
+        except OSError as e:
+            log.error("mactel failed to create %s symlink: %s", sys_dir_boot_fil, e)
+
+        sys_dir_boot_cfg = os.path.normpath(sys_dir + "boot.cfg")
+        try:
+            os.symlink(efi_cfg, sys_dir_boot_cfg)
+        except OSError as e:
+            log.error("mactel failed to create %s symlink: %s", sys_dir_boot_cfg, e)
+
+        rc = iutil.execInSysroot("hfs-bless", [bless_file])
```

Comment:
This suggests that `xen.efi` will be started, not the `boot.efi` you prepare above...